### ISSUE TITLE
2.1.1 broke uploadFile

### ISSUE THREAD

**Issue by olizh** [puppeteer/puppeteer#5527]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 2.1.1
* Platform / OS version:Mac OS 10.15.3 (19D76)
* URLs (if applicable):
* Node.js version: v12.13.0

**What steps will reproduce the problem?**
Run the following line in either headless or headful mode: 
`inputUploadHandle.uploadFile(filepath);`

**What is the expected result?**
In version 2.0.0, when uploadFile is executed, the file will be selected. You can see this if you are using headful mode. 

**What happens instead?**
However, in version 2.1.1, uploadFile results in nothing.

**Comment by renfeng** [comment]
It seems to be broken since 2.1.0

**Comment by uplg** [comment]
https://github.com/puppeteer/puppeteer/pull/5358

Also, enforcing a "change" event can resolve files input not updated in 2.1.1 but mimeType isn't ok

**Comment by olizh** [comment]
> It seems to be broken since 2.1.0

I haven't checked 2.1.0, but I've seen other people talked about 2.1.0 broke uploadFile. So it's probably broken in 2.1.0 and haven't been fixed in 2.1.1.

**Comment by DavertMik** [comment]
I downgraded to 2.0.0 but that does not solve the problem - filetype detection doesn't work there.

```
      AssertionError: expected 'application/octet-stream' to deeply equal 'image/jpeg'
      + expected - actual
      -application/octet-stream
      +image/jpeg
```

If Puppeteer team finally gets this to be fixed, I hope they also get the file type detection back.

**Comment by paullewis** [comment]
This should be fixed as of 532ae573d29063ffd081ed9ee340d71b25320dec as we changed the underlying implementation to use `DOM.setFileInputFiles`, which behaves a lot more as expected. We also fire events for `change` and `input`, too. Going to close this issue, but please let us know if it's not fixed.
